src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/DefaultRazorSemanticTokenInfoServiceTest.cs

```diff
@@ -29,16 +29,141 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.Semantic
     public class DefaultRazorSemanticTokenInfoServiceTest : DefaultTagHelperServiceTestBase
     {
         #region CSharp
+        [Fact]
+        public async Task GetSemanticTokens_CSharpBlock_HTML()
+        {
+            var txt = @$"@{{
+    var d = ""t"";
+    <p>HTML @d</p>
+}}";
+
+            var expectedData = new List<int> {
```

Comment:
lol ya we definitely need to start baseline testing these. This probably took you ages

Reply:
It... wasn't fast.

Reply:
Should keep an eye on how much time these types of efforts consume. If it makes sense to spend a day building a baselining  infrastructure for the tests then it'd probably pay itself off quick